maas/resource_maas_boot_source.go

```diff
@@ -1,12 +1,126 @@
 package maas
 
 import (
+	"context"
 	"fmt"
+	"strconv"
 
 	"github.com/canonical/gomaasclient/client"
 	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+func resourceMAASBootSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMaasBootSourceRead,
+		Description: "Provides a resource to fetch the MAAS boot source.",
+
+		Schema: map[string]*schema.Schema{
+			"created": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The creation time of the boot source.",
+			},
+			"keyring_data": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The data on the keyring for the boot source.",
+			},
+			"keyring_filename": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The filename on the keyring for the boot source.",
+			},
+			"updated": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time of most recent update of the boot source.",
+			},
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
```

Comment:
Since this is a field that we expect user to provide.

```suggestion
				Required:    true,
```